examples/functions/capture-tone-of-voice/index.ts

```diff
@@ -0,0 +1,46 @@
+import {createClient} from '@sanity/client'
+import {type DocumentEvent, documentEventHandler, type FunctionContext} from '@sanity/functions'
+
+export const handler = documentEventHandler(
+  async ({context, event}: {context: FunctionContext; event: DocumentEvent}) => {
+    // Client for the "production" dataset
+    const client = createClient({
+      ...context.clientOptions,
+      apiVersion: 'vX',
+    })
+
+    // Agent action to generate the tone of voice for the post
+    const generateToneOfVoice = await client.agent.action.generate({
+      // Set `noWrite` to `false` to write the tone of voice to the document
+      noWrite: true,
+      instructionParams: {
+        content: {
+          type: 'field',
+          path: 'body',
+        },
+      },
+      instruction: `Examine the $content. Explain the tone of voice for the post.`,
+      target: {
+        path: 'toneOfVoice',
+      },
+      conditionalPaths: {
+        defaultReadOnly: false,
+      },
+      documentId: event.data._id,
+      schemaId: '_.schemas.default',
+    })
+
+    try {
```

Comment:
it's better to wrap the whole client action operation in the try catch so that you can output the actual error, log at my examples.